diff --git a/examples/server/server.cpp b/examples/server/server.cpp
--- a/examples/server/server.cpp
+++ b/examples/server/server.cpp
@@ -838,45 +838,38 @@ int main(int argc, char ** argv) {
         // audio arrays
         std::vector<float> pcmf32;               // mono-channel F32 PCM
         std::vector<std::vector<float>> pcmf32s; // stereo-channel F32 PCM
 
+        const std::string temp_filename = generate_temp_filename(sparams.tmp_dir, "whisper-server", ".wav");
+        std::ofstream temp_file{temp_filename, std::ios::binary};
+        temp_file.write(audio_file.content.data(), static_cast<std::streamsize>(audio_file.content.size()));
+        temp_file.close();
+
         if (sparams.ffmpeg_converter) {
             // if file is not wav, convert to wav
-            // write to temporary file
-            const std::string temp_filename = generate_temp_filename(sparams.tmp_dir, "whisper-server", ".wav");
-            std::ofstream temp_file{temp_filename, std::ios::binary};
-            temp_file << audio_file.content;
-            temp_file.close();
-
             std::string error_resp = "{\"error\":\"Failed to execute ffmpeg command.\"}";
             const bool is_converted = convert_to_wav(temp_filename, error_resp, params.diarize);
             if (!is_converted) {
                 res.status = 500;
                 res.set_content(error_resp, "application/json");
+                std::remove(temp_filename.c_str());
                 return;
             }
+        }
 
-            // read audio content into pcmf32
-            if (!::read_audio_data(temp_filename, pcmf32, pcmf32s, params.diarize))
-            {
-                fprintf(stderr, "error: failed to read WAV file '%s'\n", temp_filename.c_str());
-                const std::string error_resp = "{\"error\":\"failed to read WAV file\"}";
-                res.status = 400;
-                res.set_content(error_resp, "application/json");
-                std::remove(temp_filename.c_str());
-                return;
-            }
-            // remove temp file
-            std::remove(temp_filename.c_str());
-        } else {
-            if (!::read_audio_data(audio_file.content.data(), audio_file.content.size(), pcmf32, pcmf32s, params.diarize)) {
-                fprintf(stderr, "error: failed to read audio data\n");
-                const std::string error_resp = "{\"error\":\"failed to read audio data\"}";
-                res.status = 400;
-                res.set_content(error_resp, "application/json");
-                return;
-            }
+        // read audio content into pcmf32
+        if (!::read_audio_data(temp_filename, pcmf32, pcmf32s, params.diarize))
+        {
+            fprintf(stderr, "error: failed to read WAV file '%s'\n", temp_filename.c_str());
+            const std::string error_resp = "{\"error\":\"failed to read WAV file\"}";
+            res.status = 400;
+            res.set_content(error_resp, "application/json");
+            std::remove(temp_filename.c_str());
+            return;
         }
 
+        // remove temp file
+        std::remove(temp_filename.c_str());
+
         printf("Successfully loaded %s\n", filename.c_str());
 
         // print system information
